hal/src/rtl872x/exflash_hal.cpp

```diff
@@ -564,7 +564,7 @@ int hal_exflash_uninit(void) {
 __attribute__((section(".ram.text"), noinline))
 int hal_exflash_write(uintptr_t addr, const uint8_t* data_buf, size_t data_size) {
     ExFlashLock lk;
-    CHECK(hal_flash_common_write(addr, data_buf, data_size, &perform_write, &hal_flash_common_dummy_read));
+    CHECK(hal_flash_common_write(addr, data_buf, data_size, &perform_write, &hal_flash_common_dummy_read, 0));
```

Comment:
For gen4, the only way the intermediate buffer will be used is if there is an unaligned `data_buf`?

Reply:
yes, same as before.